pipelines/pspec_pipeline/pspec_pipe.py

```diff
@@ -0,0 +1,228 @@
+#!/usr/bin/env python2
+"""
+pspec_pipe.py
+-----------------------------------------
+Copyright (c) 2018 The HERA Collaboration
+
+This script is used as the IDR2.1 power
+spectrum pipeline.
+
+See pspec_pipe.yaml for relevant parameter selections.
+"""
+import multiprocess
+import numpy as np
+import hera_cal as hc
+import hera_pspec as hp
+import hera_qm as hq
+from pyuvdata import UVData
+import pyuvdata.utils as uvutils
+import os
+import sys
+import glob
+import yaml
+from datetime import datetime
+import uvtools as uvt
+import json
+import itertools
+import aipy
+import shutil
+from collections import OrderedDict as odict
+
+
+#-------------------------------------------------------------------------------
+# Parse YAML Configuration File
+#-------------------------------------------------------------------------------
+# get config and load dictionary
+config = sys.argv[1]
+cf = hp.utils.load_config(config)
+
+# update globals with IO params, data and analysis params
+globals().update(cf['io'])
+globals().update(cf['data'])
+globals().update(cf['analysis'])
+
+# get common suffix
+data_suffix = os.path.splitext(data_template)[1][1:]
+
+# open logfile
+logfile = os.path.join(out_dir, logfile)
+if os.path.exists(logfile) and overwrite == False:
+    raise IOError("logfile {} exists and overwrite == False, quitting pipeline...".format(logfile))
+lf = open(logfile, "w")
+if joinlog:
+    ef = lf
+else:
+    ef = open(os.path.join(out_dir, errfile), "w")
+time = datetime.utcnow()
+hp.utils.log("Starting pspec pipeline on {}\n{}\n".format(time, '-'*60), f=lf, verbose=verbose)
+hp.utils.log(json.dumps(cf, indent=1) + '\n', f=lf, verbose=verbose)
+
+# Create multiprocesses
+if multiproc:
+    pool = multiprocess.Pool(nproc)
+    M = pool.map
+else:
+    M = map
+
+# change to working dir
+os.chdir(work_dir)
+
+#-------------------------------------------------------------------------------
+# Run Jacknife Data Difference
+#-------------------------------------------------------------------------------
+if run_diff:
+    # get algorithm parameters
+    globals().update(cf['algorithm']['diff'])
```

Comment:
I'm unsure about this strategy of updating `globals`. It seems a bit opaque, and I can imagine things going wrong. Perhaps we can discuss this, but probably fine to leave it for now.

Reply:
Okay, we can change to a more explicit call to the input parameter attributes